lib/features/auth/widgets/terms_and_conditions_dialog.dart

```diff
@@ -44,52 +45,50 @@ class _TermsAndConditionsDialogState extends State<TermsAndConditionsDialog> {
   @override
   Widget build(BuildContext context) {
     final locals = context.l10n;
-    final isUs = countryIso == Country.us.countryCode;
-    var termsAndConditions = locals.termsText;
+    var termsAndConditions = widget.content;
     var title = locals.fullVersionTitleTerms;
-    if (widget.typeOfTerms == TypeOfTerms.privacyPolicy) {
-      title = locals.privacyTitle;
-      termsAndConditions = locals.policyText;
+    if (widget.content == locals.termsText) {
       if (Country.unitedKingdomCodes().contains(countryIso)) {
-        termsAndConditions = locals.policyTextGb;
+        termsAndConditions = locals.termsTextGb;
       }
-      if (isUs) {
-        termsAndConditions = locals.policyTextUs;
+      if (countryIso == Country.us.countryCode) {
+        termsAndConditions = locals.termsTextUs;
       }
     }
-    if (widget.typeOfTerms == TypeOfTerms.termsAndConditions) {
+    if (widget.content == locals.policyText) {
+      title = locals.privacyTitle;
+      termsAndConditions = locals.policyText;
       if (Country.unitedKingdomCodes().contains(countryIso)) {
-        termsAndConditions = locals.termsTextGb;
+        termsAndConditions = locals.policyTextGb;
       }
-      if (isUs) {
-        termsAndConditions = locals.termsTextUs;
+      if (countryIso == Country.us.countryCode) {
+        termsAndConditions = locals.policyTextUs;
       }
     }
-    if (widget.typeOfTerms == TypeOfTerms.slimPayInfo) {
+
+    if (widget.content == locals.slimPayInfoDetail) {
       title = locals.slimPayInfoDetailTitle;
-      termsAndConditions = locals.slimPayInfoDetail;
     }
-    if (widget.typeOfTerms == TypeOfTerms.bacsInfo) {
+    if (widget.content == locals.bacsAdvanceNotice) {
       title = locals.bacsAdvanceNoticeTitle;
-      termsAndConditions = locals.bacsAdvanceNotice;
     }
-    if (widget.typeOfTerms == TypeOfTerms.directDebitGuarantee) {
+    if (widget.content == locals.bacsDdGuarantee) {
       title = locals.bacsDdGuaranteeTitle;
-      termsAndConditions = locals.bacsDdGuarantee;
     }
-    if (widget.typeOfTerms == TypeOfTerms.giftAid) {
+    if (widget.content == locals.giftAidInfoBody) {
       title = locals.giftAidInfoTitle;
-      termsAndConditions = locals.giftAidInfoBody;
     }
 
     return Theme(
       data: const FamilyAppTheme().toThemeData(),
```

Comment:
I think this is the "original" problem. We should not have the theme object here anymore. Theme switching now occurs upon logging in. Potentially you then don't need the isDarkBackground param anymore. The only place where i'm not sure if it currently works correctly is then the EmailSignupScreen.

Reply:
no I would still need the isDarkBackground, check giftAid or personal info page, the background is set independent of the theme

Reply:
But ill remove the theme object anw, you are right its not necessary